jax/experimental/jax2tf/tests/shape_poly_test.py

```diff
@@ -1855,6 +1855,19 @@ def f_jax(x):
                 lambda x: lax_control_flow.cummax(x, axis=1, reverse=False),
                 arg_descriptors=[RandArg((3, 4, 5), _f32)],
                 poly_axes=[0]),
+    PolyHarness("jnp.cumsum", "reduce_axis=poly",
+                lambda x: jnp.cumsum(x, axis=0),
+                arg_descriptors=[RandArg((3, 5), _f32)],
+                poly_axes=[0],
+                expect_error=(
+                  (None, None) if (not config.jax2tf_default_experimental_native_lowering or
```

Comment:
Question:   I do not understand why need check "native_lowering" here.  Since the error message on loops.py only tell users "avoid to run it on TPU".

Reply:
We need the check here because we want to capture the precise condition under which we get an error: native lowering and device != "TPU". Note that if I change the condition the test will fail, because an expected error is not appearing.

The error message could be more precise: you get this error if you try to use associative scan with shape polymorphism. For non-native lowering, jax2tf does not use associative scan (and jax2tf is independent of the platform). However, there is the flag `--jax2tf_associative_scan_reductions` that makes jax2tf non-native use associative scan.

With native lowering you get platform-dependent behavior, just like JAX native: will use associative scan for CPU and GPU.